functests/performance/rt-kernel.go

```diff
@@ -59,4 +60,53 @@ var _ = Describe("[performance]RT Kernel", func() {
 
 	})
 
+	It("[test_id:28526][crit:high][vendor:cnf-qe@redhat.com][level:acceptance] Non worker-cnf node should not have RT kernel installed", func() {
+		var newWorkerNode *corev1.Node
+		newRole := "worker-test"
+		newLabel := fmt.Sprintf("%s/%s", testutils.LabelRole, newRole)
+
+		By("Skipping test if cluster does not have another available worker node")
+		workerNodes, err := nodes.GetByRole(testclient.Client, "worker")
+		Expect(err).ToNot(HaveOccurred())
+
+		for _, node := range workerNodes {
+			if _, ok := node.Labels[fmt.Sprintf("%s/%s", testutils.LabelRole, testutils.RoleWorkerRT)]; ok {
+				continue
+			}
+			newWorkerNode = &node
+			break
+		}
+		if newWorkerNode == nil {
+			Skip("Skipping test because there are no additional worker nodes")
+		}
+		newWorkerNode.Labels[newLabel] = ""
+		Expect(testclient.Client.Update(context.TODO(), newWorkerNode)).ToNot(HaveOccurred())
+
+		Eventually(func() string {
+			testpod = pods.GetBusybox()
+			testpod.Namespace = testutils.NamespaceTesting
+			testpod.Spec.Containers[0].Command = []string{"uname", "-a"}
+			testpod.Spec.RestartPolicy = corev1.RestartPolicyNever
+			testpod.Spec.NodeSelector = map[string]string{
+				fmt.Sprintf("%s", newWorkerNode): "",
+			}
+
+			if err := testclient.Client.Create(context.TODO(), testpod); err != nil {
+				return ""
+			}
+
+			if err := pods.WaitForPhase(testclient.Client, testpod, corev1.PodSucceeded, 60*time.Second); err != nil {
+				return ""
+			}
+
+			logs, err := pods.GetLogs(testclient.K8sClient, testpod)
+			if err != nil {
+				return ""
+			}
+
+			return logs
+
+		}, 15*time.Minute, 30*time.Second).ShouldNot(ContainSubstring("PREEMPT RT"))
```

Comment:
Would it make sense to check if the command did run successfully, e.g by looking for "Linux" in the string?

Reply:
Ack.